### PR TITLE
Chun Yu FrontEndMentor Assignment (Rock Paper Scissors)

### PR DESCRIPTION
**Pull request by gunbux**
## Rock, Paper, Scissors, Lizard, Spock
Simple Rock Paper Scissors spinoff game made in React

### Comparison Images
`Original`
desktop-step-1-bonus
`Implementation`
image
`Original`
desktop-step-3-bonus
`Implementation`
image
`Original`
mobile-step-1-bonus
`Implementation`
image
`Original`
mobile-step-4-bonus
`Implementation`
image

---

### CODE REVIEW

File: src/components/RockPaperScissors/SelectState.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { BottomSelect, MiddleSelect, Overlay, SelectContainer, TopSelect } from './styles/SelectState.styled'
+import pentagonIcon from '../../assets/RockPaperScissors/bg-pentagon.svg'
+import { Option } from '../../pages/RockPaperScissors/util'
+
+type Props = {
+  renderButton: (option: Option) => ReactElement
+  optionList: Option[]
+}
+
+const SelectState = ({ renderButton, optionList }: Props) => (
```
Comment (limwenfeng) [inline src/components/RockPaperScissors/SelectState.tsx:11]: You might want to change name of the props renderButton as it is not an actual button.

File: src/components/RockPaperScissors/PickContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { Option, optionList } from '../../pages/RockPaperScissors/util'
+import { Blank, PickText, StyledPickContainer } from './styles/PickContainer.styled'
+
+type Props = {
+  header: string
+  selection: string
+  renderOption: (option: Option) => ReactElement
+}
+
+const PickContainer = ({ header, selection, renderOption }: Props) => {
```
Comment (xinyee20) [inline src/components/RockPaperScissors/PickContainer.tsx:11]: consider just writing it as 'props: Props' ? so there won't be a long list of parameter

File: src/components/RockPaperScissors/ResultState.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+type Props = {
+  renderOption: (option: Option) => ReactElement
+  result: string
+  handleReset: () => void
+  currentSelection: string
+  comSelection: string
+}
+
+const ResultState = ({ renderOption, result, handleReset, currentSelection, comSelection }: Props) => {
```
Comment (xinyee20) [inline src/components/RockPaperScissors/ResultState.tsx:21]: similarly, consider just writing it as 'props: Props' ? it's more accident proof too in cases like this where there is a higher chance a prop may be missed out

File: src/components/RockPaperScissors/PickState.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import { Option } from '../../pages/RockPaperScissors/util'
+import PickContainer from './PickContainer'
+
+type Props = {
+  currentSelection: string
+  comSelection: string
+  renderOption: (option: Option) => ReactElement
+}
+
+const PickState = ({ currentSelection, comSelection, renderOption }: Props) => {
```
Comment (xinyee20) [inline src/components/RockPaperScissors/PickState.tsx:12]: similarly, consider just writing it as 'props: Props' ?

File: src/components/RockPaperScissors/SelectState.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { BottomSelect, MiddleSelect, Overlay, SelectContainer, TopSelect } from './styles/SelectState.styled'
+import pentagonIcon from '../../assets/RockPaperScissors/bg-pentagon.svg'
+import { Option } from '../../pages/RockPaperScissors/util'
+
+type Props = {
+  renderOption: (option: Option) => ReactElement
+  optionList: Option[]
+}
+
+const SelectState = ({ renderOption, optionList }: Props) => (
```
Comment (xinyee20) [inline src/components/RockPaperScissors/SelectState.tsx:11]: similarly, consider just writing it as 'props: Props' ?

File: src/components/RockPaperScissors/PickState.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <PickStateContainer>
+      <PickContainer renderOption={renderOption} header="YOU PICKED" selection={currentSelection} />
+      <PickContainer renderOption={renderOption} header="THE HOUSE PICKED" selection={comSelection} />
+
+      {/*{renderPickContainer('YOU PICKED', currentSelection)}*/}
+      {/*{renderPickContainer('THE HOUSE PICKED', comSelection)}*/}
```
Comment (xinyee20) [inline src/components/RockPaperScissors/PickState.tsx]: unused code can be removed

File: src/components/RockPaperScissors/styles/Scoreboard.styled.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components'
+
+export const StyledIconTitle = styled.img<{ imgSize?: string }>`
+  height: ${(props) => props.imgSize ?? '2rem'};
+  width: ${(props) => props.imgSize ?? '2rem'};
+`
+export const StyledScoreboard = styled.div`
+  width: 100%;
+  max-width: 800px;
+  //min-height: 150px;
```
Comment (xinyee20) [inline src/components/RockPaperScissors/styles/Scoreboard.styled.ts]: delete if not used?

File: src/components/RockPaperScissors/styles/Scoreboard.styled.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components'
+
+export const StyledIconTitle = styled.img<{ imgSize?: string }>`
+  height: ${(props) => props.imgSize ?? '2rem'};
+  width: ${(props) => props.imgSize ?? '2rem'};
+`
+export const StyledScoreboard = styled.div`
+  width: 100%;
```
Comment (xinyee20) [inline src/components/RockPaperScissors/styles/Scoreboard.styled.ts]: can consider lowering it to maybe 85% and add margin so there is still some spacing between the scoreboard and the sides of the screen (will improve mobile view)

File: src/components/RockPaperScissors/styles/Scoreboard.styled.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+  max-width: 800px;
+  //min-height: 150px;
+  border-radius: 5px;
+  border: 2px solid #4e5f7d;
```
Comment (xinyee20) [inline src/components/RockPaperScissors/styles/Scoreboard.styled.ts]: border can be increased slightly to maybe 3 and border-radius can be increase to maybe around 10px to look more like the original?

File: src/pages/RockPaperScissors/styles/RockPaperScissors.styled.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import styled from 'styled-components'
+
+export const MainContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  min-width: 100vw;
```
Comment (xinyee20) [inline src/pages/RockPaperScissors/styles/RockPaperScissors.styled.ts]: I think this line is causing the horizontal scrolling, can consider removing it

**Review by xinyee20 — COMMENTED** [review]
- Consider putting your file within FrontendMentor folder instead of under pages folder to improve organisation of the files
- Font has been provided for the score text and number, can be added in if you want it to look closer to the original
  font-weight: 600; (text) / 700; (number)
  font-family: 'Barlow Semi Condensed';
  color: hsl(229, 25%, 31%); (number)
*score number size can increase a bit more
- When the player won/lose/draw, the results doesn't come up if I don't click the screen. Maybe it will be better to show the results immediately without needing any action from the user?

**Comment by gunbux** [comment]
> * Consider putting your file within FrontendMentor folder instead of under pages folder to improve organisation of the files
> * Font has been provided for the score text and number, can be added in if you want it to look closer to the original
>   font-weight: 600; (text) / 700; (number)
>   font-family: 'Barlow Semi Condensed';
>   color: hsl(229, 25%, 31%); (number)
>   *score number size can increase a bit more
> * When the player won/lose/draw, the results doesn't come up if I don't click the screen. Maybe it will be better to show the results immediately without needing any action from the user?

- Refactored code to be in FrontendMentor folder
- Changed Font to look more like the original (the method to import fonts in the notion wasn't working, used @import at index.css instead)
- increased score number size
- set timer to automatically switch to results after 1000ms

**Comment by limwenfeng** [comment]
LGTM!

**Review by limwenfeng — APPROVED** [review]

**Pull request merged**